### ISSUE TITLE
Python 3.5 compatibility

### ISSUE THREAD

**Issue by warsaw** [malthe/chameleon#206]
Chameleon appears to be incompatible with Python 3.5:

```
$ tox -e py35
GLOB sdist-make: /home/barry/projects/debian/chameleon/upstream/setup.py
py35 create: /home/barry/projects/debian/chameleon/upstream/.tox/py35
py35 inst: /home/barry/projects/debian/chameleon/upstream/.tox/dist/Chameleon-2.23.dev0.zip
py35 installed: Chameleon==2.23.dev0
py35 runtests: PYTHONHASHSEED='3917267219'
py35 runtests: commands[0] | python setup.py test -q
/home/barry/projects/debian/chameleon/upstream/.tox/py35/lib/python3.5/site-packages/setuptools/dist.py:285: UserWarning: Normalizing '2.23-dev' to '2.23.dev0'
  normalized_version,
running test
running egg_info
writing top-level names to src/Chameleon.egg-info/top_level.txt
writing src/Chameleon.egg-info/PKG-INFO
writing dependency_links to src/Chameleon.egg-info/dependency_links.txt
reading manifest file 'src/Chameleon.egg-info/SOURCES.txt'
reading manifest template 'MANIFEST.in'
writing manifest file 'src/Chameleon.egg-info/SOURCES.txt'
running build_ext
Traceback (most recent call last):
  File "setup.py", line 82, in <module>
    'benchmark': Benchmark,
  File "/usr/lib/python3.5/distutils/core.py", line 148, in setup
    dist.run_commands()
  File "/usr/lib/python3.5/distutils/dist.py", line 955, in run_commands
    self.run_command(cmd)
  File "/usr/lib/python3.5/distutils/dist.py", line 974, in run_command
    cmd_obj.run()
  File "/home/barry/projects/debian/chameleon/upstream/.tox/py35/lib/python3.5/site-packages/setuptools/command/test.py", line 146, in run
    self.with_project_on_sys_path(self.run_tests)
  File "/home/barry/projects/debian/chameleon/upstream/.tox/py35/lib/python3.5/site-packages/setuptools/command/test.py", line 127, in with_project_on_sys_path
    func()
  File "/home/barry/projects/debian/chameleon/upstream/.tox/py35/lib/python3.5/site-packages/setuptools/command/test.py", line 167, in run_tests
    testRunner=self._resolve_as_ep(self.test_runner),
  File "/usr/lib/python3.5/unittest/main.py", line 93, in __init__
    self.parseArgs(argv)
  File "/usr/lib/python3.5/unittest/main.py", line 140, in parseArgs
    self.createTests()
  File "/usr/lib/python3.5/unittest/main.py", line 147, in createTests
    self.module)
  File "/usr/lib/python3.5/unittest/loader.py", line 219, in loadTestsFromNames
    suites = [self.loadTestsFromName(name, module) for name in names]
  File "/usr/lib/python3.5/unittest/loader.py", line 219, in <listcomp>
    suites = [self.loadTestsFromName(name, module) for name in names]
  File "/usr/lib/python3.5/unittest/loader.py", line 190, in loadTestsFromName
    return self.loadTestsFromModule(obj)
  File "/home/barry/projects/debian/chameleon/upstream/.tox/py35/lib/python3.5/site-packages/setuptools/command/test.py", line 36, in loadTestsFromModule
    tests.append(self.loadTestsFromName(submodule))
  File "/usr/lib/python3.5/unittest/loader.py", line 190, in loadTestsFromName
    return self.loadTestsFromModule(obj)
  File "/home/barry/projects/debian/chameleon/upstream/.tox/py35/lib/python3.5/site-packages/setuptools/command/test.py", line 22, in loadTestsFromModule
    tests.append(TestLoader.loadTestsFromModule(self, module))
  File "/usr/lib/python3.5/unittest/loader.py", line 123, in loadTestsFromModule
    tests.append(self.loadTestsFromTestCase(obj))
  File "/usr/lib/python3.5/unittest/loader.py", line 92, in loadTestsFromTestCase
    loaded_suite = self.suiteClass(map(testCaseClass, testCaseNames))
  File "/usr/lib/python3.5/unittest/suite.py", line 24, in __init__
    self.addTests(tests)
  File "/usr/lib/python3.5/unittest/suite.py", line 57, in addTests
    for test in tests:
  File "/home/barry/projects/debian/chameleon/upstream/src/chameleon/tests/test_doctests.py", line 10, in __new__
    return getattr(self, test)()
  File "/home/barry/projects/debian/chameleon/upstream/src/chameleon/tests/test_doctests.py", line 40, in test_compiler
    compiler, optionflags=OPTIONFLAGS)
  File "/usr/lib/python3.5/doctest.py", line 2365, in DocTestSuite
    tests = test_finder.find(module, globs=globs, extraglobs=extraglobs)
  File "/usr/lib/python3.5/doctest.py", line 923, in find
    self._find(tests, obj, name, module, source_lines, globs, {})
  File "/usr/lib/python3.5/doctest.py", line 982, in _find
    if ((inspect.isroutine(inspect.unwrap(val))
  File "/usr/lib/python3.5/inspect.py", line 471, in unwrap
    while _is_wrapper(func):
  File "/usr/lib/python3.5/inspect.py", line 465, in _is_wrapper
    return hasattr(f, '__wrapped__')
  File "/home/barry/projects/debian/chameleon/upstream/src/chameleon/utils.py", line 25, in __getattr__
    return _ast.__dict__.get(name) or getattr(_ast, self.aliases[name])
KeyError: '__wrapped__'
ERROR: InvocationError: '/home/barry/projects/debian/chameleon/upstream/.tox/py35/bin/python setup.py test -q'
___________________________________ summary ___________________________________
ERROR:   py35: commands failed
```

**Comment by malthe** [comment]
Let me have a look. Thanks for reporting.

**Comment by malthe** [comment]
Fixed in 7fa34663bd06dda5c75872f11b5af4dd602e2be4. This is released as Chameleon==2.24.

**Comment by warsaw** [comment]
Looks great, thanks for the quick turnaround!
